packages/chaire-lib-backend/src/models/db/default.db.queries.ts

```diff
@@ -92,7 +104,11 @@ const createMultiple = async <T extends GenericAttributes, U>(
 
         const chunkSize = 250;
 
-        return await knex.batchInsert(tableName, _newObjects as any, chunkSize).returning(_returning);
+        const query = knex.batchInsert(tableName, _newObjects as any, chunkSize).returning(returning);
+        if (options.transaction) {
+            query.transacting(options.transaction);
+        }
+        return await query;
```

Comment:
If you await and something bad happens, the try catch in this function will catch it. If you return without the await, then it's the caller's try/catch that will be responsible to catch the error, wherever that is. Since we want to sort of control the exceptions that are thrown by database queries (ie have our own exceptions thrown and not knex's), then the queries need to await.

That said, with the current rewrite, the try/catch are in the public user-facing functions, so those need to await, but maybe not the inner private functions, where there is no try/catch.